### ISSUE TITLE
Using with Rollup

### ISSUE THREAD

**Issue by smddzcy** [josdejong/mathjs#1232]
We're currently having problems using this library with Rollup. Here's the error:

![screen shot 2018-09-02 at 23 10 22](https://user-images.githubusercontent.com/13895224/44960294-a142b180-af05-11e8-8300-370947e75ea9.png)

And my Rollup config:

```
export default {
  external: ['react', 'react-dom'],
  input: 'src/index.ts',
  output: [
    {
      file: pkg.main,
      name: 'index',
      format: 'umd',
      sourcemap: false
    },
    {
      file: pkg.module,
      format: 'es',
      sourcemap: false
    }
  ],
  watch: {
    include: 'src/**'
  },
  plugins: [
    // Node built-in fns support
    builtins(),
    postcss({
      extensions: ['.css'],
      plugins: [url({ url: 'inline' })],
      inject: false
    }),
    // Allow json resolution
    json(),
    // Compile TypeScript files
    typescript({
      useTsconfigDeclarationDir: true,
      abortOnError: false,
      check: false
    }),
    // Allow node_modules resolution, so you can use 'external' to control
    // which external modules to include in the bundle
    // https://github.com/rollup/rollup-plugin-node-resolve#usage
    resolve({
      jsnext: true,
      main: true
    }),
    // Allow bundling cjs modules (unlike webpack, rollup doesn't understand cjs)
    commonjs({
      namedExports: {
        '../adaptableblotter/node_modules/react/index.js': [
          'Children',
          'Component',
          'PropTypes',
          'createElement',
          'cloneElement'
        ],
        '../adaptableblotter/node_modules/react-dom/index.js': ['render']
      },
      sourceMap: false
    }),

    // Resolve source maps to the original source
    // sourceMaps(),
  ]
}
```

Am I doing something wrong or is it related to your exports?

**Comment by josdejong** [comment]
Thanks for reporting.

I don't have much experience with rollup. I just spend 20 minutes trying to get a simple rollup config working but I can't even get a function imported and bundled from lodash (messing around with [rollup-plugin-node-resolve](https://github.com/rollup/rollup-plugin-node-resolve)). I'm sorry.

I would like to try in a simple stand alone setup without TypeScript to be sure whether the issue originates in TypeScript or rollup.

**Comment by josdejong** [comment]
Thanks. Maybe importing a bundle from `dist` works, so everything is included.

**Comment by benrodenhaeuser** [comment]
For me, the following works: 

- install Math.js via npm 
- use the `commonjs` and `node-resolve` rollup plugins

Then you can do:

```javascript
import * as math from 'mathjs';
```

**Comment by josdejong** [comment]
Thanks for your input. It can very well be that this issue is fixed in v6, which moved from commonjs to ES6 modules. @smddzcy can you give v6 a try?

**Comment by EliasHasle** [comment]
I fail to import mathjs with Rollup too. (I am posting my problem here, as the name of the issue matches, and the issue is open and otherwise not alive.)

For some scripts (typed-function.js among them) `this` is `undefined`, and (probably as a consequence) `typed-function` does not export. My rollup setup applies `rollup-plugin-node-resolve` and exports in _umd_ format.

'I have tried multiple ways of importing, including:

`import {create, FibonacciHeapDependencies} from "mathjs";
const {FibonacciHeap} = create({FibonacciHeapDependencies});`

`import {create, all} from "mathjs";
const math = create(all);`

`import * as math from "mathjs"
const FibonacciHeap = math.type.FibonacciHeap;`

All of them fail.

**Comment by josdejong** [comment]
@EliasHasle thanks for reporting. Can you try 

```js
import {create, all} from "mathjs/main/es6/";
const math = create(all);
```
explicitly pointing to the ES6 code?

**Comment by EliasHasle** [comment]
I now tried:
````
import {create, FibonacciHeapDependencies} from "mathjs/main/ES6";
const {FibonacciHeap} = create({FibonacciHeapDependencies});
````

with `rollup-plugin-commonjs` disabled, and got the samme errors as before.

**Comment by josdejong** [comment]
This issue is most likely fixed in `v6.0.4`, can you give this a try @EliasHasle and @smddzcy?

**Comment by EliasHasle** [comment]
I can confirm that my issue is resolved with 6.0.4. Thanks! :smile

**Comment by josdejong** [comment]
That's great to hear, thanks for the feedback. 

Since multiple people report they can successfully use mathjs in their rollup projects I'll close this issue now.
